Mathlib/Tactic/Positivity/Basic.lean

```diff
@@ -446,3 +447,29 @@ def evalRatDen : PositivityExt where eval {u α} _ _ e := do
     assumeInstancesCommute
     pure $ .positive q(den_pos $a)
   | _, _ => throwError "not Rat.num"
+
+/-- Extension for `posPart`. `a⁺` is always nonegative, and positive if `a` is. -/
+@[positivity _⁺]
+def evalPosPart : PositivityExt where eval zα pα e := do
+  match e with
+  | ~q(@posPart _ $instαlat $instαgrp $a) =>
+    assertInstancesCommute
+    match ← catchNone (core zα pα a) with
+    | .positive pf => return .positive q(posPart_pos $pf)
+    | _ => return .nonnegative q(posPart_nonneg $a)
+  | _ => throwError "not `posPart`"
+
+/-- Extension for `negPart`. `a⁻` is always nonegative. -/
+@[positivity _⁻]
+def evalNegPart : PositivityExt where eval _ _ e := do
+  match e with
+  | ~q(@negPart _ $instαlat $instαgrp $a) =>
+    assertInstancesCommute
+    return .nonnegative q(negPart_nonneg $a)
+  | _ => throwError "not `negPart`"
+
+-- There seems to be a bug in `Positivity.core` that makes it fail (instead of returning `.none`) in
+-- the first example
+example (a : ℤ) : 0 ≤ a⁺ := by positivity
```

Comment:
I think the convention is that these examples should go in `test/positivity.lean`.